Userland/Applications/PartitionEditor/PartitionModel.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022, Samuel Bowman <sam@sambowman.tech>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Applications/PartitionEditor/PartitionModel.h>
+#include <LibPartition/EBRPartitionTable.h>
+#include <LibPartition/GUIDPartitionTable.h>
+#include <LibPartition/MBRPartitionTable.h>
+
+namespace PartitionEditor {
+
+String PartitionModel::column_name(int column) const
+{
+    switch (column) {
+    case Column::Partition:
+        return "Partition";
+    case Column::StartBlock:
+        return "Start Block";
+    case Column::EndBlock:
+        return "End Block";
+    default:
+        VERIFY_NOT_REACHED();
+    }
+}
+
+GUI::Variant PartitionModel::data(GUI::ModelIndex const& index, GUI::ModelRole role) const
+{
+    if (role != GUI::ModelRole::Display)
+        return {};
+    if (!m_partition_table)
+        return {};
+
+    auto optional_partition = m_partition_table->partition(index.row());
+    if (optional_partition.has_value()) {
+        auto partition = optional_partition.release_value();
+
+        switch (index.column()) {
+        case Column::Partition:
+            return index.row() + 1;
+        case Column::StartBlock:
+            return partition.start_block();
+        case Column::EndBlock:
+            // FIXME: Either MBR end block is off by one (if supposed to be exclusive bound)
+            //        or GPT end block is off by one (if supposed to be inclusive bound).
+            //        This is an issue in LibPartition.
+            return partition.end_block();
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    }
+
+    return {};
+}
+
+ErrorOr<void> PartitionModel::set_device_path(String path)
```

Comment:
I don't think we need to take this by value if we don't move it anyway

Reply:
Good catch. Fixed.